### PR TITLE
leave network on reciept of DeleteHost action

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
leave network on receipt of deleteHost action.  Ensures that zombie node will be deleted on server.

## Provide link to Netmaker PR if required
n/a
## Provide testing steps
see GRA-1456
## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [ ] Netclient & Netmaker are awesome.

### CODE REVIEW

**Review by Aceix — APPROVED** [review]

File: functions/mqhandlers.go [hunk 1]
```diff
@@ -224,12 +224,16 @@ func HostUpdate(client mqtt.Client, msg mqtt.Message) {
 		}
 		restartDaemon = true
 	case models.DeleteHost:
-		clearRetainedMsg(client, msg.Topic())
-		unsubscribeHost(client, serverName)
-		deleteHostCfg(client, serverName)
-		config.WriteNodeConfig()
-		config.WriteServerConfig()
-		resetInterface = true
+		faults, err := LeaveNetwork(hostUpdate.Node.Network, true)
```
Comment (abhishek9686) [inline functions/mqhandlers.go:227]: why are we leaving the network here on the delete host prompt?

Comment (mattkasun) [reply functions/mqhandlers.go:227]: So the server will get the message and delete the zombie node

Comment (abhishek9686) [reply functions/mqhandlers.go:227]: doesn't that happen when you delete a node already?

**Review by theguy951357 — APPROVED** [review]
checklist on kitemaker passes
critical checklist passes

**Pull request closed without merging**